### DISCUSSION TITLE
rpmbuild error : Directory not found

### DISCUSSION THREAD

**Discussion by rmehtacs** [rpm-software-management/rpm#2074] [Q&A] [answered]
the rpmbuild is giving error directory not found. Below is the error

**Processing files: product-X.Y.Z-ABC.i686
error: Directory not found: /home/build/builds/product-dev/build/Linux-i686-SLES11/Release/product_stage/var/tools/licenses/***

However directory exists.

Why searching  the error, some one has suggested solution for similar issue that this could be because buildroot is not set properly. 
For this rpm build the buildroot is specified in custom provided spec files as below

**%define buildroot /home/build/builds/product-dev/build/Linux-i686-SLES11/Release/product_stage**
and product_root is defined as 
**%define product_root /var/tools**

I tried by adding %buildroot to rpmmacros but that didn't works for me.

I am using RPM v4.14 (SLES-15).

This code works perfectly fine with RPM v4.4.2.3 (SLES-11)

**Comment by rmehtacs** [comment]
@pmatilai Please guide me.

**Comment by voxik** [comment]
This report is missing lot of context leaving everybody to guess what you tried to achieve. If you provided .spec file at minimum or better SRPM, you would have higher chances to get any response.

**Reply by rmehtacs** [reply]
For some reason I can't provide spec or SRPMS.
The spec file contains below lines (which I believe is causing the error)
%files -f files.list
........
........
%attr(0604, root, root) %dir %{product_root}/licenses/*
........
........

**Comment by pmatilai** [comment, marked as answer]
Buildroot is no longer user defineable in rpm >= 4.6, ie the rpm versions released in past 13 or so years.
Running into this is an indication of abusing rpm: in rpm packaging, *rpmbuild* is supposed to handle the build of the software from sources to binaries, whereas you're pointing it to on-disk binaries built by some other means.

Rpm will not prevent you from copying files from an arbitrary location to the per-build buildroot though, so that's how you can work around it.
